Cubical/Homotopy/Group/Pi4S3/BrunerieSelfcontained.agda

```diff
@@ -0,0 +1,111 @@
+{-# OPTIONS --safe #-}
+module Cubical.Homotopy.Group.Pi4S3.BrunerieSelfcontained where
+
+open import Cubical.Core.Primitives
+
+open import Cubical.Foundations.Prelude using (_,_ ; fst ; snd ; _≡_ ; transport ; _∙_ ; refl ; isProp→isSet ; J)
+open import Cubical.Foundations.Equiv using (_≃_ ; isEquiv ; isPropIsEquiv ; idEquiv)
+open import Cubical.Foundations.Pointed using (Pointed₀ ; pt ; _→∙_)
+open import Cubical.Foundations.HLevels using (hSet ; hGroupoid ; isOfHLevelTypeOfHLevel ; isOfHLevelPath ; isOfHLevelΠ)
+open import Cubical.Foundations.Univalence using (Glue ; ua)
+
+open import Cubical.Data.Int using (ℤ ; pos ; neg ; isSetℤ)
+
+open import Cubical.HITs.S1 using (S¹ ; base ; loop ; helix ; rotIsEquiv)
+open import Cubical.HITs.S2 using (S² ; base ; surf ; S²∙ ; S¹×S¹→S² ; S²ToSetElim)
+open import Cubical.HITs.Join.Base using (join ; inl ; inr ; push)
+open import Cubical.HITs.SetTruncation renaming (∥_∥₂ to ∥_∥₀ ; ∣_∣₂ to ∣_∣₀ ; squash₂ to squash₀ ; rec to rec₀) using ()
+open import Cubical.HITs.GroupoidTruncation renaming (∥_∥₃ to ∥_∥₁ ; ∣_∣₃ to ∣_∣₁ ; squash₃ to squash₁ ; rec to rec₁) using ()
+open import Cubical.HITs.2GroupoidTruncation renaming (∥_∥₄ to ∥_∥₂ ; ∣_∣₄ to ∣_∣₂ ; squash₄ to squash₂ ; rec to rec₂ ; elim to elim₂) using ()
+open import Cubical.HITs.Susp using (Susp ; north ; south ; merid ; Susp∙ ; toSusp)
+
+S¹∙ : Pointed₀
+S¹∙ = (S¹ , base)
+
+∥_∥₁∙ ∥_∥₂∙ : Pointed₀ → Pointed₀
+∥ A , a ∥₁∙ = ∥ A ∥₁ , ∣ a ∣₁
+∥ A , a ∥₂∙ = ∥ A ∥₂ , ∣ a ∣₂
+
+Ω Ω² : Pointed₀ → Pointed₀
+Ω (_ , a) = ((a ≡ a) , refl)
+Ω² p = Ω (Ω p)
+
+HopfS² : S² → Type₀
+HopfS² base = S¹
+HopfS² (surf i j) = Glue S¹ (λ { (i = i0) → _ , idEquiv S¹
+                               ; (i = i1) → _ , idEquiv S¹
+                               ; (j = i0) → _ , idEquiv S¹
+                               ; (j = i1) → _ , _ , rotIsEquiv (loop i) } )
+
+codeS² : S² → hGroupoid _
+codeS² s = ∥ HopfS² s ∥₁ , squash₁
+
+codeTruncS² : ∥ S² ∥₂ → hGroupoid _
+codeTruncS² = rec₂ (isOfHLevelTypeOfHLevel 3) codeS²
+
+encodeTruncS² : Ω ∥ S²∙ ∥₂∙ .fst → ∥ S¹ ∥₁
+encodeTruncS² p = transport (λ i → codeTruncS² (p i) .fst) ∣ base ∣₁
+
+codeS¹ : S¹ → hSet _
+codeS¹ s = ∥ helix s ∥₀ , squash₀
+
+codeTruncS¹ : ∥ S¹ ∥₁ → hSet _
+codeTruncS¹ = rec₁ (isOfHLevelTypeOfHLevel 2) codeS¹
+
+encodeTruncS¹ : Ω ∥ S¹∙ ∥₁∙ .fst → ∥ ℤ ∥₀
+encodeTruncS¹ p = transport (λ i → codeTruncS¹ (p i) .fst) ∣ pos 0 ∣₀
+
+g8 : Ω² ∥ S²∙ ∥₂∙ .fst → Ω ∥ S¹∙ ∥₁∙ .fst
+g8 p i = encodeTruncS² (p i)
+
+g9 : Ω ∥ S¹∙ ∥₁∙ .fst → ∥ ℤ ∥₀
+g9 = encodeTruncS¹
+
+-- Use trick to eliminate away the truncation last
```

Comment:
@ecavallo  I wonder if this trick of yours makes any difference... What do you think?

Reply:
So g8/g9 could potentially land-in/start-from `Ω S¹`